### PR TITLE
features: setup data source

### PR DESCRIPTION
**Pull request by PanVova**

### CODE REVIEW

File: core/build.gradle
Comment (NickHuk) [inline core/build.gradle]: Why did you add spaces? Make sure your android studio codestyle is square-android

File: dataSource1/src/main/java/QuizGateway.kt
Comment (NickHuk) [inline dataSource1/src/main/java/QuizGateway.kt]: if you place this files to `datasource` package please name it ...Datasource instead of Gateway

File: repositoryImpl/build.gradle
Comment (NickHuk) [inline repositoryImpl/build.gradle]: why have you renamed this?

File: dataSource1/src/main/java/QuizGateway.kt
Comment (NickHuk) [inline dataSource1/src/main/java/QuizGateway.kt]: `category`, `difficulty`, `type` should be enums

File: dataSourceImpl/src/main/java/SharedPreferencesGatewayImpl.kt
Comment (NickHuk) [inline dataSourceImpl/src/main/java/SharedPreferencesGatewayImpl.kt]: It's a wrong structure, Repository should depend on gateway, not vice versa. Dependencies order: PreferenceRepository -> SharedPreferenceGateway -> SharedPreferences;

File: dataSourceImpl/src/main/java/model/PreferencesRepository.kt
Comment (NickHuk) [inline dataSourceImpl/src/main/java/model/PreferencesRepository.kt]: It's better to use [jetpack datastore](https://developer.android.com/topic/libraries/architecture/datastore?gclid=Cj0KCQiAhf2MBhDNARIsAKXU5GT-EIWjFhULziJ-E3JW_Msz-wx3w4q3U1NTWOGBNH7o7aba0SI3KWoaAs3pEALw_wcB&gclsrc=aw.ds) instead of shared preferences

File: dataSourceImpl/src/main/java/model/PreferencesRepository.kt
Comment (NickHuk) [inline dataSourceImpl/src/main/java/model/PreferencesRepository.kt]: Use [sharedPreference ktx](https://developer.android.com/kotlin/ktx?gclid=Cj0KCQiAhf2MBhDNARIsAKXU5GRwF1rq0nV82pF-hjnGrTHoFAS9o-QrWHrNm9W4CblLvH1parVHi1QaAvyjEALw_wcB&gclsrc=aw.ds)

File: dataSourceImpl/src/main/java/model/PreferencesRepository.kt
Comment (NickHuk) [inline dataSourceImpl/src/main/java/model/PreferencesRepository.kt]: Return nullable token instead of using `!!`

File: dataSource1/src/main/java/QuizGateway.kt
Comment (NickHuk) [inline dataSource1/src/main/java/QuizGateway.kt]: Create separate gateway for token as questions and  token are logically different things and cannot be provided by same datasource

File: dataSource1/src/main/java/model/Question.kt
Comment (NickHuk) [inline dataSource1/src/main/java/model/Question.kt]: Use camelCase - for variable names

File: dataSource1/src/main/java/SharedPreferencesGateway.kt
Comment (NickHuk) [inline dataSource1/src/main/java/SharedPreferencesGateway.kt]: use kotlin approach 
interface
```
var token: String
```
implementation
```
overrid var token: String
  get() = ...
  set(value) = ...
```

File: dataSourceImpl/src/main/java/QuizGatewayImpl.kt
Comment (NickHuk) [inline dataSourceImpl/src/main/java/QuizGatewayImpl.kt]: Avoid creating clases with empty implementations. In this case just remove this file if it has empty implementations

**Review by NickHuk — CHANGES_REQUESTED** [review]

File: dataSource1/src/main/java/QuizGateway.kt
Comment (PanVova) [reply dataSource1/src/main/java/QuizGateway.kt]: +

File: repositoryImpl/build.gradle
Comment (PanVova) [reply repositoryImpl/build.gradle]: changed it back

File: dataSource1/src/main/java/QuizGateway.kt
Comment (PanVova) [reply dataSource1/src/main/java/QuizGateway.kt]: How? Do I need to make two models or just wait until you merge other PR so I can use EnumClasses

File: dataSourceImpl/src/main/java/QuizGatewayImpl.kt
Comment (PanVova) [reply dataSourceImpl/src/main/java/QuizGatewayImpl.kt]: +

**Pull request merged**
